catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/model/FilterNodeAssertionSupport.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.forms.model;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.core.Is.is;
+
+import org.codice.ddf.catalog.ui.forms.model.pojo.FilterLeafNode;
+import org.codice.ddf.catalog.ui.forms.model.pojo.FilterNode;
+import org.codice.ddf.catalog.ui.forms.model.pojo.FilterTemplatedLeafNode;
+
+/** As more test cases are added, more support functions will be needed. */
+class FilterNodeAssertionSupport {
+  private FilterNodeAssertionSupport() {}
+
+  static void assertParentNode(FilterNode node, String expectedType, int expectedChildCount) {
+    assertParentNode(node, expectedType);
+    assertThat(node.getNodes(), hasSize(expectedChildCount));
+  }
+
+  static void assertParentNode(FilterNode node, String expectedType) {
+    assertThat(node.getType(), is(expectedType));
+    assertThat(node.getNodes(), notNullValue());
+    assertThat(node.isLeaf(), is(false));
+  }
+
+  static void assertLeafNode(
+      FilterNode node, String expectedType, String expectedProperty, String expectedValue) {
+    assertThat(node.getType(), is(expectedType));
+    assertThat(node.getNodes(), is(nullValue()));
+    assertThat(node.isLeaf(), is(true));
+    assertThat(FilterLeafNode.class.isInstance(node), is(true));
```

Comment:
✏️ Alternatively, `assertThat(node, isA(FilterLeafNode.class))`

Reply:
The type restrictions on the generic method make this call impossible (and thus redundant). It expects a `Matcher<T>` based upon the `Class<T>` that is passed as an argument. It can support "safe" assertions meant for verifying obvious truths: 
```
assertThat(node, isA(Object.class));
```
but provides no utility for downcasting. The following can't compile: 
```
assertThat(node, isA(FilterLeafNode.class));
```